### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Kaiserabbas**
# Decorate a class

- Created a class Nameable.
  - Implemented a method called correct_name that will raise a NotImplementedError.
  - Turned Person class to Nameable.
  - Person class inherits from Nameable.
  - This class has a method correct_name. It simply returns the name attribute.

- Created a Base Decorator 
  - It inherits from Nameable.
  - In the constructor assigned a nameable object from params to an instance variable.
  - Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
  
- Prepare CapitalizeDecorator and TrimmerDecorator
  - For the CapitalizeDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
  - For the TrimmerDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -1,14 +1,24 @@
-class Person
+require_relative 'Nameable'
```
Comment (zeff96) [inline src/person.rb]: - [ ] Hello! 👋 It's essential to pay attention to the naming conventions for relative path files in your project. To ensure that your project runs smoothly as expected, please make sure that the file names are all in lowercase. This is important for consistency and compatibility with various systems and platforms. 🚀

```suggestion
require_relative 'nameable'
```

File: src/base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'Nameable'
```
Comment (zeff96) [inline src/base_decorator.rb]: - [ ] Hello! 👋 It's essential to pay attention to the naming conventions for relative path files in your project. To ensure that your project runs smoothly as expected, please make sure that the file names are all in lowercase. This is important for consistency and compatibility with various systems and platforms. 🚀

```suggestion
require_relative 'nameable'
```

File: src/nameable.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "Subclasses must implement the 'correct_name' method."
```
Comment (zeff96) [inline src/nameable.rb]: - [ ] Hello, When it comes to best coding practices, it's highly recommended to include both the class and the specific method within that class that triggered the error. 🚫🐞 This practice serves as a valuable aid for efficient code debugging. 🕵️‍♂️🔍 By doing so, you create a clear and organized trail that helps you pinpoint the source of issues quickly. This not only accelerates the debugging process but also enhances the overall maintainability and readability of your code. 📈💻 So, remember to name those culprits when errors occur! 📣😄

```suggestion
    class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end

```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @Kaiserabbas , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Descriptive pull request
:heavy_check_mark: Correct workflow
:heavy_check_mark: Linters passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/nameable.rb (hunk 3 above)
Comment (Kaiserabbas) [reply src/nameable.rb]: Thanks for the review, considering the points raised above I have changed my code accordingly.

**Review by Alusp — APPROVED** [review]
Hi @Kaiserabbas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats
## Highlights

- [ ] Implement composition. 👍
- [ ] Use Professional Readme file 👍
- [ ] Descriptive PR Summary used 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alusp in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**